python/tvm/relay/op/nn/nn.py

```diff
@@ -1471,6 +1471,47 @@ def bias_add(data, bias, axis=1):
     return _make.bias_add(data, bias, axis)
 
 
+def matmul(data, weight, units=None, out_dtype="", data_transposed=False, weight_transposed=False):
+    """Matmul operator.
+    Applies a linear transformation. The X & W can be transposed.
+
+    .. math::
+
+        `Y = X * W`
+
+    Parameters
+    ----------
+    data : tvm.relay.Expr
+        The input data to the operator,
+        of shape `(d_1, d_2, ..., d_n, units_in)` or `(d_1, d_2, ..., units_in, d_n)`.
```

Comment:
Shouldn't both input shapes by dimension 2?

Reply:
No, the input of matmul is supposed to be a multiple-dim tensor(not limited to 2). This is copied from the original `nn.dense`.

Other frameworks like Pytorch also has such definition.

Reply:
Can you update the definition of the computation above to reflect these shapes then?